src/main/java/org/kohsuke/github/GitHub.java

```diff
@@ -482,7 +483,34 @@ void check(String apiUrl) throws IOException {
      * Otherwise this method throws {@link IOException} to indicate the problem.
      */
     public void checkApiUrlValidity() throws IOException {
-        retrieve().to("/", GHApiInfo.class).check(apiUrl);
+        try {
+            retrieve().to("/", GHApiInfo.class).check(apiUrl);
+        } catch (IOException ioe) {
+            if (isPrivateModeEnabled()) {
```

Comment:
The method basically is the same but now we have included a new case.

I tested with:

1. https://api.github.com/
1. http://github.mycompany.com/api/v3/
1. https://github.mycompany.com/api/v3/ (self-signed certificate)
1. https://github.mycompany.com/api/v3/ (certificate)